### ISSUE TITLE
[CODEX] K26 — baseline ML evaluation on expanded diagnostic dataset

### ISSUE THREAD

**Issue by an89537171644** [an89537171644/GBK#26]
Task code: K26

Title: Baseline ML evaluation on expanded diagnostic dataset

Goal:
Evaluate baseline non-neural ML on the expanded K25 diagnostic dataset. K26 must not add a neural network and must not make ML a design checker. ML remains advisory-only and deterministic SP63 checks remain mandatory.

Scope:
- extend the baseline ML report with an expanded diagnostic classification block;
- evaluate `overall_status` classification on diagnostic dataset limit 100;
- use only non-neural baseline models such as DummyClassifier and LogisticRegression or other simple sklearn models already supported by the project;
- report classification metrics: accuracy, macro_f1, per-class precision/recall, confusion matrix, class distribution, train/test rows;
- keep target leakage controls by excluding direct status targets, failure_reason, warning_text, and target status columns from feature sets;
- support `python -m sp63_core ml-baseline --diagnostic-limit 100 --json`;
- document that deterministic-derived features can leak deterministic outcomes and are review-only;
- update tests and documentation.

Forbidden changes:
- do not change `checks/bending.py`;
- do not change `checks/shear.py`;
- do not change `checks/cracking.py`;
- do not change `checks/crack_width.py`;
- do not change `checks/deflection.py`;
- do not change calculation formulas;
- do not change material values;
- do not change `validation/external.py`;
- do not add a neural network;
- do not implement UI or Streamlit;
- do not add full SP 63 text;
- do not add personal, grant, or private documents.

Required implementation:
- update `src/sp63_core/ml/baseline_report.py`;
- add `expanded_diagnostic_classification` report block;
- include input-only and deterministic-derived feature modes if practical;
- use fixed random_state for train/test evaluation;
- warn if diagnostic dataset is small or lacks group_key for leakage checking;
- update CLI payload as needed;
- update tests and documentation.

Required checks:
- `python -m pytest`
- `ruff check .`
- `python -m sp63_core validate --golden`
- `python -m sp63_core validate --generate-dataset-limit 20 --json`
- `python -m sp63_core materials-audit --json`
- `python -m sp63_core manual-cases --json`
- `python -m sp63_core ml-readiness --generate-dataset-limit 100 --json`
- `python -m sp63_core diagnostic-dataset --limit 100 --json`
- `python -m sp63_core ml-readiness --diagnostic --json`
- `python -m sp63_core ml-baseline --diagnostic-limit 100 --json`

Do not merge automatically. Do not enable auto-merge.
